src/viperleed/utilities/poscar/modify_vacuum.py

```diff
@@ -12,31 +12,52 @@
 __license__ = 'GPLv3+'
 
 from copy import deepcopy
+from dataclasses import dataclass
 import logging
 
 import numpy as np
 
-from viperleed.calc.classes.slab.errors import (NotEnoughVacuumError,
+from viperleed.calc.classes.slab.errors import (
+                                                NotEnoughVacuumError,
+                                                VacuumError,
                                                 WrongVacuumPositionError)
 from viperleed.utilities.poscar.base import _PoscarStreamCLI
 
 
 logger = logging.getLogger(__name__)
 
 
-def modify_vacuum(slab, vacuum_gap_size, absolute=False):
+@dataclass
+class VacuumGapInfo:
+    """Information about a slab's desired vacuum gap.
+
+    Attributes
+    ----------
+    size : float
+        Size or size change of the desired vacuum gap, in angstrom.
+    absolute : bool, optional
+        Whether `size` should be considered as the absolute value of
+        the new vacuum gap rather than its change. Default is False.
+    accept_small_gap : bool, optional
+        Whether the script should tolerate a resulting vacuum
+        gap smaller than the minimum value for a viperleed.calc
+        run, as long as it is non-negative. Default is False.
+    """
+
+    size: float
+    absolute: bool = False
+    accept_small_gap: bool = False
+
+
+def modify_vacuum(slab, vacuum_gap_info):
```

Comment:
I cannot comment on L69 below, but please update the documentation of what this function raises and when.